plugins/janus_recordplay.c

```diff
@@ -606,6 +644,21 @@ static const char *janus_recordplay_parse_codec(const char *dir, const char *fil
 				json_t *f = json_object_get(info, "f");
 				if(f && json_is_string(f) && fmtp && fmtplen > 0)
 					g_snprintf(fmtp, fmtplen, "%s", json_string_value(f));
+				/* Any extmaps? */
+				json_t *ext_json = json_object_get(info, "x");
+				if(ext_json && json_is_array(ext_json) && json_array_size(ext_json) > 0) {
+					size_t ext_n = json_array_size(ext_json), i = 0;
+					for(i = 0; i < ext_n; i++) {
+						const char *ext_str = json_string_value(json_array_get(ext_json, i));
+						char delim = ':', **ext_split = g_strsplit(ext_str, &delim, 2);
+						janus_sdp_attribute *ext = janus_sdp_attribute_create(
+							g_strdup(ext_split[0]),
+							ext_split[1] ? g_strdup(ext_split[1]) : NULL);
```

Comment:
This is wrong. It's assuming `g_strsplit` will succeed, that there are at least two parts and that both are not `NULL`. This will segfault if any of those assumptions are incorrect. Besides, there's no need to `g_strdup` when doing `janus_sdp_attribute_create`, as that method already allocates new stuff for those strings: you're generating memory leaks, as those strings you're allocating are lost.

Reply:
I've slightly misread g_strsplit documentation, thanks for pointing this out.